lib/util/quarantine/oaep.js

```diff
@@ -0,0 +1,203 @@
+//    checking. typically, you'd error if the first byte of the input buffer is
+//    anything other than 0x0, if the first db digestlen bytes mismatch a digest
+//    of the label (which by default is empty/null), or if any of the db bytes from
+//    there to The One (0x01) is anything but 0x0.
+//
+//    but we know exactly what our cleartext length should be. we aren't trying
+//    to write a consumer-use library here: if it's not right the subsequent AES
+//    decryption will fail anyway, and if we should have rejected the user input
+//    it would have been upon data ingress which could have been many moons ago.
+//
+// ultimately, there are four basic things to convince yourself of here to verify
+// the security of this operation (though if you are a real security expert you
+// should ignore me):
+//
+// 1. constant time execution.
+// 2. no branch instructions based off secret values.
+// 3. all allocation up front.
+// 4. all allocated buffers filled fully with input data.
+//
+// the deal with #4 is that Buffer.allocUnsafe is analogous to malloc.
+// (Buffer.alloc is analogous to calloc.)
+//
+// okay, so that's all. please don't use this. bye.
+//
+// (** so it additionally transpires that Java allows the specification of
+// OAEPWithSHA256AndMGF1Padding OR OAEPWithSHA-256AndMGF1Padding which doesn't
+// seem like so big a deal until you learn that without a hyphen, Bouncy Castle
+// is used, and with a hyphen Java JCE is used. and /that/ doesn't seem like so
+// big a deal until you learn that BC uses SHA256 for both OAEP and MGF1, while
+// JCE uses SHA256 for OEAP, but sticks with SHA1 for MGF1. IETF thinks you
+// should not mix the algos.)
+//
+// WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
+// WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
+// WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
+// WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
+// WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
+////////////////////////////////////////////////////////////////////////////////
+
+
+////////////////////////////////////////////////////////////////////////////////
+// EXTERNAL FUNCTION ANALYSIS
+// some analysis and justification of external functions used.
+//
+// Math.ceil (v8)
+// is constant time under numbers smaller than 0x80000000 (in which case a cast
+// to uint32 is performed.) additionally, the only values it may be fed are
+// 32 / 32 = 1, and 223 / 32 = 6.96875, neither of which are sensitive.
+//
+// createHash/update/digest
+// i sure hope these are cryptographically safe.
+//
+// Buffer.allocUnsafe, Buffer#subarray
+// some branching etc occurs internally but all input values are constant.
+//
+// Buffer#writeInt32BE
+// does some branching (checkInt in buffer.js) but the input sequence is fixed:
+// 0 0 1 2 3 4 5 6
```

Comment:
I had to think through that double 0. So it's only used for the mask generation and the first 0 is for the seed then it's the db sequence.

Reply:
yes. i think there's some really stripped down version of this where you actually just unmask the seed and then unmask the final 32 bytes of the data block and that saves a lot of work and looping but it's nice for the algorithm to actually look like the algorithm.